MekHQ/src/mekhq/MekHqXmlUtil.java

```diff
@@ -660,4 +663,32 @@ public static String getEntityNameFromXmlString(Node node) {
         String model = attrs.getNamedItem("model").getTextContent();
         return chassis + " " + model;
     }
+
+    /**
+     * Parses a node for the money information it contains handling
+     * different save versions
+     * @param node node that contains the money information
+     * @return a money object
+     */
+    public static Money getMoneyFromXmlNode(Node node) {
+        // Check the XML data version
+        if (node.hasAttributes()) {
+            Node attribute = node.getAttributes().getNamedItem("version");
+            if (attribute != null && attribute.getTextContent().trim().equals("2")) {
```

Comment:
Ideally, versioning should be at the root level, and then all loading routines would need to handle versioning (if they want). I just don't know the whole thing good enough to do that. I'll see if I can figure out something better.

But, do we care about the size of the CPNX file? I mean, we are already using xml instead of something more compact like json, so is this much of a concern?

Reply:
CPNX load time is a real issue for some of our users. They have 300+ mb files, so I'm somewhat sensitive to anything that makes the file larger.

I think a strategy that checks for a space, and if none found presumes this to be the old version. Otherwise it parses as the new version.

Reply:
Simple enough, I'll change it to that.